components/ContentSearch/index.js

```diff
@@ -51,30 +53,37 @@ const ContentSearch = ({ onSelectItem, placeholder, label, contentTypes, mode, e
 		setSearchString(keyword);
 		setIsLoading(true);
 
-		const searchQuery = `wp/v2/search/?search=${keyword}&subtype=${contentTypes.join(
-			',',
-		)}&type=${mode}`;
-
-		apiFetch({
-			path: searchQuery,
-		}).then((results) => {
-			const newResults = results.filter((result) => {
-				let keep = true;
-
-				if (excludeItems.length) {
-					excludeItems.forEach((item) => {
-						if (item.id === result.id) {
-							keep = false;
-						}
-					});
-				}
-
-				return keep;
-			});
-
-			setSearchResults(newResults);
+		if (searchCache[keyword]) {
+			setSearchResults(searchCache[keyword]);
 			setIsLoading(false);
-		});
+		} else {
+			const searchQuery = `wp/v2/search/?search=${keyword}&subtype=${contentTypes.join(
+				',',
+			)}&type=${mode}`;
+
+			apiFetch({
+				path: searchQuery,
+			}).then((results) => {
+				const newResults = results.filter((result) => {
+					let keep = true;
```

Comment:
Since this is async, we should bail here if the component has unmounted. This can be achieved with `useRef` and an effect hook somewhere at the top of the component.

```JS
const mounted = useRef(true);

	useEffect(() => {
		return () => {
			mounted.current = false;
		};
	}, []);
```

That effect hook's return callback will set `mounted` to false when the component is destroyed. Then if `mounted.current` is false, we can bail out of the `fetch` callback before setting state.